### PR TITLE
Debug mode

### PR DESCRIPTION
**Pull request by vreynolds**
## Which problem is this PR solving?

- allow users to read API responses in the lambda function logs
- adding HONEYCOMB_DEBUG env var that needs to be configured on the lambda function
  - another possibility is to configure debug at the template level, but my assumption is that users would prefer to just update the environment variable on the lambda function. Even if that's not the case, this is an easier starting point, and we can add a debug toggle to the templates in the future if a need emerges.
- updates #71 

## Short description of the changes

- basically lifted from the [lambda extension](https://github.com/honeycombio/honeycomb-lambda-extension/blob/4cf403391d0441bcb5761f2d2e68b99455d0201c/main.go#L44)

### CODE REVIEW

**Review by JamieDanielson — APPROVED** [review]
🐞
